File: clj/beispiel_5.clj
(import (java_mag Beispiel_5 Beispiel_5$Hund Beispiel_5$Katze))
(println (Beispiel_5/istEin (Beispiel_5$Hund.)))
(println (Beispiel_5/istEin (Beispiel_5$Katze.)))
(doseq [x [(Beispiel_5$Hund.) (Beispiel_5$Katze.)]]
  (println (Beispiel_5/istEin x)))

File: src/java_mag/Beispiel_5.java
package java_mag;

import java.util.Arrays;

public class Beispiel_5 {

	public interface Tier {
	}

	public static class Hund implements Tier {
	}

	public static class Katze implements Tier {
	}

	public static String istEin(Tier h) {
		return "Tier";
	}

	public static String istEin(Hund h) {
		return "Hund";
	}

	public static String istEin(Katze k) {
		return "Katze";
	}

	public static void main(String[] args) {
		System.out.println(istEin(new Hund()));
		System.out.println(istEin(new Katze()));
		for (Tier x : Arrays.asList(new Hund(), new Katze()))
			System.out.println(istEin(x));
	}
}
